// fetch1231231/insert.js
export function insert() {
    document.getElementById('formData').addEventListener('submit', function (e) {
      e.preventDefault();
  
      const id = document.getElementById('id').value;
      const title = document.getElementById('title').value;
      const views = document.getElementById('views').value;
  
      const messageElement = document.getElementById('message');
      
      if (id == '' || title == '' || views == '') {
        messageElement.textContent = 'Please fill out all fields.';
        return;
      }
  
      fetch('http://localhost:3000/posts', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          id: id,
          title: title,
          views: views
        }),
      })
      .then(response => {
        if (!response.ok) {
          throw new Error('Network response was not ok');
        }
        return response.json();
      })
      .then(data => {
        messageElement.textContent = 'Data inserted successfully!';
        console.log('Data inserted:', data);
      })
      .catch(error => {
        console.error('Error occurred:', error);
        messageElement.textContent = 'Error inserting data. Please try again.';
      });
    });
  }
